### PR TITLE
docker: add docker scripts for single node validator

### PR DESCRIPTION
**Pull request by d-roak**
`NODE_NAME=name docker compose -f docker/compose/validator.docker-compose.yml up -d`

### CODE REVIEW

File: docker/scripts/validator.entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+	"pubkey": $VALIDATOR_KEY,
+	"amount": "$STAKING_AMOUNT",
+	"moniker": $MONIKER,
+	"commission-rate": "0.1",
+	"commission-max-rate": "0.2",
+	"commission-max-change-rate": "0.01",
+	"min-self-delegation": "1",
+    "website": "https://droak.sh",
+    "security": "me+aligned@droak.sh"
```
Comment (JulianGCalderon) [inline docker/scripts/validator.entrypoint.sh:44]: We should leave an empty website and security by default.

**Review by JulianGCalderon — CHANGES_REQUESTED** [review]
The docker-compose works as intended, but i noticed that the create-validator transaction is sent right after the node is setup. 

The node won't be able to validate blocks until it has finished syncing up with the blockchain, which means that (if the blockchain is large enough) it could be slashed even before starting to validate.

Is there a way to wait for the node to catch-up before sending the create-validator transaction?

File: docker/scripts/validator.entrypoint.sh (hunk 1 above)
Comment (d-roak) [reply docker/scripts/validator.entrypoint.sh:44]: ohhh mb, on that, it was extracted from my node config

**Comment by ManuelBilbao** [comment]
There's an endpoint which we can use for checking if syncing is complete: `localhost:26657/status`.

Maybe we can add something like this before the `create-validator` order:
```sh
while [ $(curl -s localhost:26657/status |  jq '.result.sync_info.catching_up') == "true" ]
do
    sleep 1
done
```

**Review by tomyrd — CHANGES_REQUESTED** [review]

File: multi_node_setup.sh [hunk 2]
```diff
@@ -74,9 +72,7 @@ for (( i=1; i <= "$#"; i++ )); do
     if [ $((i+1)) -le "$#" ]; then
         j=$((i+1))
         cp prod-sim/${!i}/config/genesis.json prod-sim/${!j}/config/genesis.json
-    else
-        cp prod-sim/${!i}/config/genesis.json prod-sim/$1/config/genesis.json
-    fi
```
Comment (tomyrd) [inline multi_node_setup.sh:79]: `else` referenced in [this comment](https://github.com/yetanotherco/aligned_layer_tendermint/pull/82#discussion_r1532638687).

**Review by tomyrd — CHANGES_REQUESTED** [review]

**Review by ManuelBilbao — CHANGES_REQUESTED** [review]
If you stop the container and then start it again, the `node-setup` and `validator-setup` are run again, generating a new key with different address, thus generating another validator

File: multi_node_setup.sh [hunk 3]
```diff
@@ -56,16 +56,14 @@ for (( i=1; i <= "$#"; i++ )); do
...
```
[the remark this answers is not in the document]
Comment (d-roak) [reply multi_node_setup.sh:61]: in the case `i=1` u have `cp prod-sim/$1/config/genesis.json prod-sim/$1/config/genesis.json` that is doing nothing. Same applies on the other scenario

File: docker/scripts/validator.entrypoint.sh (hunk 1 above)
Comment (d-roak) [reply docker/scripts/validator.entrypoint.sh]: fixed in https://github.com/yetanotherco/aligned_layer_tendermint/pull/82/commits/6819e224238fd907c40e2e1a792ad311e4f47903

**Comment by d-roak** [comment]
> The node won't be able to validate blocks until it has finished syncing up with the blockchain, which means that (if the blockchain is large enough) it could be slashed even before starting to validate.

Not sure how are cosmos chains handle status, but @ManuelBilbao suggestion should work to fix it

> If you stop the container and then start it again, the node-setup and validator-setup are run again, generating a new key with different address, thus generating another validator

True. A possible alternative is to go with `NODE_NAME=name docker compose -f docker/compose/validator.docker-compose.yml up validator-runner` (and remove the depends_on). To make it easier for operators, you can wrap it in the `Makefile` or a bash script

I opened this PR with what I had on my side to make it work for my specific environment and generic enough for it to work everywhere else. I don't intend to make it super reliable and fine-tuned for generic use cases. So you can either:
- work on top of it (only available for maintainers)
- merge it and work from there
- create a new PR with whatever you find useful and close this one 🫡

cc @JulianGCalderon @ManuelBilbao @tomyrd @MauroToscano

File: multi_node_setup.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (d-roak) [reply multi_node_setup.sh:61]: image

this is the message that you get on those `else` cases

[the remark this answers is not in the document]
Comment (tomyrd) [reply multi_node_setup.sh:61]: Ohh you are right. In the case `i=1` this else makes it fail. In other scenarios though that copies goes from the last to the first. Thanks for the help.

**Comment by MauroToscano** [comment]
We can merge this and keep working on it

**Review by ManuelBilbao — APPROVED** [review]

**Review by JulianGCalderon — APPROVED** [review]
LGTM

**Review by tomyrd — APPROVED** [review]
LGTM

**Pull request merged**